src/include/duckdb/common/fast_mem.hpp

```diff
@@ -0,0 +1,692 @@
+//                         DuckDB
+//
+// duckdb/common/fast_mem.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/types.hpp"
+
+template <size_t SIZE>
+static inline void MemcpyFixed(void *dest, const void *src) {
+	memcpy(dest, src, SIZE);
+}
+
+template <size_t SIZE>
+static inline int MemcmpFixed(const void *str1, const void *str2) {
+	return memcmp(str1, str2, SIZE);
+}
+
+namespace duckdb {
+
+//! This templated memcpy is significantly faster than std::memcpy,
+//! but only when you are calling memcpy with a const size in a loop.
+//! For instance `while (<cond>) { memcpy(<dest>, <src>, const_size); ... }`
+static inline void FastMemcpy(void *dest, const void *src, const size_t size) {
+	D_ASSERT(size % 8 == 0);
```

Comment:
If size%8 has to be 0, do we need to add the cases for size=1, size=2, etc?

Reply:
This can be removed but I forgot it was there, my bad. We may want to use this code elsewhere where the assertion does not hold.